src/watertap_contrib/seto/solar_models/surrogate/trough/data/training_trough_surrogate.py

```diff
@@ -73,8 +77,18 @@ def create_rbf_surrogate(
     return rbf_surr
 
 
-def get_training_validation(dataset_filename, n_samples, training_fraction):
+def get_training_validation(
+    dataset_filename, n_samples, training_fraction, heat_load_range, hours_storage_range
+):
```

Comment:
Should we have some check on heat_load_range and hours_storage_range? What happens if the user enters something invalid? Do we have a test to check that as well?

Reply:
This is a good question. I would defer to @dguittet , but I am not sure we need a test for this.

I don't think this file is tested directly. I view this as sort of a demo/how-to for creating a trough surrogate. Presumably, the user would know the limits of their system/surrogate, something that we could not know a priori for a test (unless there are practical limits to these variables, but that is a different consideration maybe).

To me, this really relates to how we want to implement surrogate generation moving forward, and how flexible and automated we want that implementation to be moving forward.

Reply:
As the code stands now, it looks like we're trying to grant a tad more flexibility in that the user can specify their own ranges specifically for heat load and heat storage hours if they so choose to, rather than always being bound to specific ranges that were defined for a single location. Of course, they'd be doing this at their own risk, and our documentation should reflect that down the line. 

However, since we have the functionality already in play here, whether the test is implemented directly or indirectly through the `test_trough_surrogate file`, my only suggestion is that we test what happens when using this function with an (1) alternative set of define ranges, simply to check that the functionality performs as intended (rather than going through the full workflow to create a new surrogate), and (2) add something in this function that will raise an exception if the user provides input that is unexpected and invalid, and test that this exception is raised when entering something nonsensical as input.

Reply:
I should add one thing--I was assuming that the function here would be intended for use to arrive at alternative surrogates while also serving as an example. Not clear whether this was meant to serve as both or simply to just serve as a representative example.

Reply:
Added an error check